sys-utils/hwclock.c

```diff
@@ -608,9 +608,8 @@ set_hardware_clock_exact(const struct hwclock_control *ctl,
 	}
 
 	newhwtime = sethwtime
-		    + (int)(time_diff(nowsystime, refsystime)
-			    - delay /* don't count this */
-			    + 0.5 /* for rounding */);
+		    + lround(time_diff(nowsystime, refsystime)
+			    - delay /* don't count this */);
```

Comment:
Not sure if remove +0.5 is a good idea. Would it be possible that lround() returns time in the past?

Reply:
maybe ceil() is better?

Reply:
ceil() sounds good.